runelite-client/src/main/java/net/runelite/client/graphics/ModelOutlineRenderer.java

```diff
@@ -0,0 +1,1073 @@
+						int x = pixelPos % imageWidth;
+						int y = pixelPos / imageWidth;
+						if (x < clipX1 || x >= clipX2 ||
+							y < clipY1 || y >= clipY2 ||
+							visited[pixelPos] == currentVisitedNumber)
+						{
+							continue;
+						}
+						visited[pixelPos] = currentVisitedNumber;
+
+						imageData[pixelPos] = color;
+
+						if (pixelPos % imageWidth != 0)
+						{
+							outlinePixels[nextDistArrayPosX][outlinePixelsLengths[nextDistArrayPosX]++] = pixelPos - 1;
+						}
+						if ((pixelPos + 1) % imageWidth != 0)
+						{
+							outlinePixels[nextDistArrayPosX][outlinePixelsLengths[nextDistArrayPosX]++] = pixelPos + 1;
+						}
+						outlinePixels[nextDistArrayPosY][outlinePixelsLengths[nextDistArrayPosY]++] = pixelPos - imageWidth;
+						outlinePixels[nextDistArrayPosY][outlinePixelsLengths[nextDistArrayPosY]++] = pixelPos + imageWidth;
+					}
+				}
+			}
+			else
+			{
+				for (int i2 = 0; i2 < outlinePixelsLengths[distArrayPos]; i2++)
+				{
+					int pixelPos = outlinePixels[distArrayPos][i2];
+					int x = pixelPos % imageWidth;
+					int y = pixelPos / imageWidth;
+					if (x < clipX1 || x >= clipX2 ||
+						y < clipY1 || y >= clipY2 ||
+						visited[pixelPos] == currentVisitedNumber)
+					{
+						continue;
+					}
+					visited[pixelPos] = currentVisitedNumber;
+
+					imageData[pixelPos] =
+						((((color & 0xFF0000) * alpha + (imageData[pixelPos] & 0xFF0000) * (255 - alpha)) / 255) & 0xFF0000) +
+						((((color & 0xFF00) * alpha + (imageData[pixelPos] & 0xFF00) * (255 - alpha)) / 255) & 0xFF00) +
+						((((color & 0xFF) * alpha + (imageData[pixelPos] & 0xFF) * (255 - alpha)) / 255) & 0xFF);
+
+					if (pixelPos % imageWidth != 0)
+					{
+						outlinePixels[nextDistArrayPosX][outlinePixelsLengths[nextDistArrayPosX]++] = pixelPos - 1;
+					}
+					if ((pixelPos + 1) % imageWidth != 0)
+					{
+						outlinePixels[nextDistArrayPosX][outlinePixelsLengths[nextDistArrayPosX]++] = pixelPos + 1;
+					}
+					outlinePixels[nextDistArrayPosY][outlinePixelsLengths[nextDistArrayPosY]++] = pixelPos - imageWidth;
+					outlinePixels[nextDistArrayPosY][outlinePixelsLengths[nextDistArrayPosY]++] = pixelPos + imageWidth;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Draws an outline around a model to an image
+	 *
+	 * @param localX The local x position of the model
+	 * @param localY The local y position of the model
+	 * @param localZ The local z position of the model
+	 * @param orientation The orientation of the model
+	 * @param outlineWidth The width of the outline
+	 * @param innerColor The color of the pixels of the outline closest to the model
+	 * @param outerColor The color of the pixels of the outline furthest away from the model
+	 */
+	private void drawModelOutline(Model model,
+		int localX, int localY, int localZ, int orientation,
+		int outlineWidth, Color innerColor, Color outerColor)
+	{
+		if (outlineWidth <= 0)
+		{
+			return;
+		}
+
+		if (isReset)
+		{
+			scheduler.registerObject(this);
```

Comment:
I would probably move this logic to RuneLite class and call it from there (by injecting this)